pkg/controller/atlasdeployment/atlasdeployment_controller_test.go

```diff
@@ -854,9 +860,12 @@ func TestReconciliation(t *testing.T) {
 		sch.AddKnownTypes(v1.GroupVersion, &v1.AtlasBackupScheduleList{})
 		sch.AddKnownTypes(v1.GroupVersion, &v1.AtlasBackupPolicy{})
 		sch.AddKnownTypes(v1.GroupVersion, &v1.AtlasDatabaseUserList{})
+		// Subresources need to be explicitly set now since controller-runtime 1.15
+		// https://github.com/kubernetes-sigs/controller-runtime/issues/2362#issuecomment-1698194188
 		k8sClient := fake.NewClientBuilder().
 			WithScheme(sch).
 			WithObjects(secret, project, bPolicy, bSchedule, deployment).
+			WithStatusSubresource(bPolicy, bSchedule).
```

Comment:
Is this just for tracking the resources (controlled by the same controller: `AtlasDeployment`) status?